src/main/webapp/script.js

```diff
@@ -12,7 +12,105 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function dummyDBMentee(name, email, username, school, classStanding, intro, reason, desiredSkills) {
+    this.name = name;
+    this.email = email;
+    this.username = username;
+    this.school = school;
+    this.classStanding = classStanding;
+    this.intro  = intro;
+    this.reason = reason;
+    this.desiredSkills = desiredSkills;
+}
 
+function dummyDBMentor(name, email, username, job, college, intro, expertise) {
+    this.name = name;
+    this.email = email;
+    this.username = username;
+    this.job = job;
+    this.college = college;
+    this.intro  = intro;
+    this.expertise = expertise;
+}
 
+/*
+Function: addTableCards()
+Adapts the feed to the username whether is a mentor or mentee to show the right information
+    
+Notes:
+    Up until now works in a local way with structs containing dummy info 
+    Next step is to adapt it to work with datastorage
+*/
+function addTableCards() {
 
+    /*Could be an option to obtain the value from the redirection from other pages, still in progress...
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const username = urlParams.get('username');*/
 
+    const username = "john123";
+
+    var myMentees = [
+        new dummyDBMentee('John S', 'john@myemail.com', 'john123', 'mySchool', 2021, "I'm a student", "I like to learn", "C++"),
+        new dummyDBMentee('Mary S', 'mary@myemail.com', 'mary123', 'mySchool', 2022, "I'm a student", "I like to learn", "Java"),
+        new dummyDBMentee('Nick S', 'nick@myemail.com', 'nick123', 'mySchool', 2024, "I'm a student", "I like to learn", "Go")
+    ];
+
+    var myMentors = [
+        new dummyDBMentor('John T', 'john@myemail.com', 'john124', 'Senior Software Engineer', 'MyCollege', "I'm a software engineer", "JavaScript"),
+        new dummyDBMentor('Mary T', 'mary@myemail.com', 'mary124', 'UX Designer', 'MyCollege', "I'm a UX specialist", "Accesibility")
+    ];
+
+    //Having a mentor username, showing list of mentees
```

Comment:
Maybe I'm missing something, but it looks like you're showing both mentors and mentees, but I think you only want to show one or the other depending on the user's status.

I would split this function into two separate ones, one for mentors and one for mentees. This will also make it easier when choosing which function to call.

Or what you can do is still have the "addTableCards" as your top function, but within it, separate the code into two other functions, perhaps something like "addMenteeCards" and "addMentorCards".

Reply:
You're right, it would be easier to separate it into two different functions and just call them.